flow/include/flow/Traceable.h

```diff
@@ -0,0 +1,247 @@
+/*
+ * Traceable.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FLOW_TRACEABLE_H
+#define FLOW_TRACEABLE_H
+#pragma once
+
+#include <stdarg.h>
+#include <stdint.h>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+#define PRINTABLE_COMPRESS_NULLS 0
+
+template <class IntType>
+char base16Char(IntType c) {
+	switch ((c % 16 + 16) % 16) {
+	case 0:
+		return '0';
+	case 1:
+		return '1';
+	case 2:
+		return '2';
+	case 3:
+		return '3';
+	case 4:
+		return '4';
+	case 5:
+		return '5';
+	case 6:
+		return '6';
+	case 7:
+		return '7';
+	case 8:
+		return '8';
+	case 9:
+		return '9';
+	case 10:
+		return 'a';
+	case 11:
+		return 'b';
+	case 12:
+		return 'c';
+	case 13:
+		return 'd';
+	case 14:
+		return 'e';
+	case 15:
+		return 'f';
+	default:
+		std::abort();
```

Comment:
I changed this from an assert to an abort

Reply:
Out of curiosity, what is the reason for this change?

Reply:
Our `ASSERT` macro isn't available here, so this seemed like a simple alternative